pkg/blockbuilder/blockbuilder.go

```diff
@@ -239,3 +240,308 @@ func (b *BlockBuilder) getLagForPartition(ctx context.Context, partition int32)
 
 	return kadm.GroupMemberLag{}, lastErr
 }
+
+type partitionState struct {
+	Partition          int32
+	Lag                int64
+	Commit             kadm.Offset
+	CommitRecTimestamp time.Time
+	SeenTillOffset     int64
```

Comment:
nits on naming: till is the short version of until, why not use Until? Also maybe something to make it more obvious if this offset was consumed or not - `latestConsumedOffset` or `oldestSeenOffset`, `firstUnconsummedOffset`?